build.sbt

```diff
@@ -10,9 +10,9 @@ autoScalaLibrary := false
 // More compiler warnings
 scalacOptions ++= Seq("-deprecation", "-unchecked", "-feature", "-Xlint")
 
-javacOptions ++= Seq("-source", "1.6", "-target", "1.6")
+javacOptions ++= Seq("-source", "8", "-target", "8")
```

Comment:
Are we ready to require that Java clients use Java 8 or higher?

Reply:
I don't know what workflows would break there, that's why I was considering a major version change for the client. I'm open to ideas though, if we make it 100% compatible with 1.6, the public API would look nowhere near as clean going forwards.

Reply:
Bumping the major version sounds fine to me, per semver. A cleaner API sounds good!